common/management/src/role/role_mgr.rs

```diff
@@ -0,0 +1,191 @@
+// Copyright 2021 Datafuse Labs.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::sync::Arc;
+
+use common_exception::ErrorCode;
+use common_exception::Result;
+use common_exception::ToErrorCode;
+use common_meta_api::KVApi;
+use common_meta_types::GrantObject;
+use common_meta_types::IntoSeqV;
+use common_meta_types::MatchSeq;
+use common_meta_types::MatchSeqExt;
+use common_meta_types::OkOrExist;
+use common_meta_types::Operation;
+use common_meta_types::RoleInfo;
+use common_meta_types::SeqV;
+use common_meta_types::UpsertKVAction;
+use common_meta_types::UserPrivilegeSet;
+
+use crate::role::role_api::RoleMgrApi;
+
+static ROLE_API_KEY_PREFIX: &str = "__fd_roles";
+
+pub struct RoleMgr {
+    kv_api: Arc<dyn KVApi>,
+    role_prefix: String,
+}
```

Comment:
There is an issue with such a impl that @ariesdevil has been working on.

`kv_api` need to renew the handshake info if the backend metasrv restarted.
Because handshake info is not persistent on metasrv.

`kv_api`(and `meta_api`) has to discover a handshake error then resend the handshake request.

@ariesdevil will soon provide a solution to this problem.